R/chat-structured.R

```diff
@@ -53,20 +53,30 @@ convert_from_type <- function(x, type) {
     } else if (S7_inherits(type@items, TypeEnum)) {
       factor(as.character(x), levels = type@items@values)
     } else if (S7_inherits(type@items, TypeObject)) {
-      cols <- lapply(names(type@items@properties), function(name) {
-        vals <- lapply(x, function(y) y[[name]])
-        convert_from_type(vals, type_array(type@items@properties[[name]]))
-      })
-      names(cols) <- names(type@items@properties)
-      list2DF(cols)
+      if (type@items@additional_properties) {
+        x
```

Comment:
If we're preserving the order from the type when `additional_properties = FALSE`, should we do the same when additional properties are allowed? I think that might be nice, but I'm not convinced it would be necessary, but I noticed that we're testing this in the `FALSE` case.